[CI] ClassificationIT testWithOnlyTrainingRowsAndTrainingPercentIsHundred failing

**Build Scans:**
- [elasticsearch-periodic #6101 / openjdk22_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/uzr3rjpflegoe)
- [elasticsearch-periodic #6050 / openjdk17_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/dcibak66eqwwk)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.ml.integration.ClassificationIT.testWithOnlyTrainingRowsAndTrainingPercentIsHundred" -Dtests.seed=3C5CEA9E4BCC7CA9 -Dtests.locale=sr-Latn-XK -Dtests.timezone=Asia/Barnaul -Druntime.java=22
```

**Applicable branches:**
8.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.ClassificationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testWithOnlyTrainingRowsAndTrainingPercentIsHundred),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Stats were: {"id":"classification_only_training_data_and_training_percent_is_100","state":"started","progress":[{"phase":"reindexing","progress_percent":100},{"phase":"loading_data","progress_percent":100},{"phase":"feature_selection","progress_percent":100},{"phase":"coarse_parameter_search","progress_percent":100},{"phase":"fine_tuning_parameters","progress_percent":93},{"phase":"final_training","progress_percent":0},{"phase":"writing_results","progress_percent":0},{"phase":"inference","progress_percent":0}],"data_counts":{"training_docs_count":300,"test_docs_count":0,"skipped_docs_count":0},"memory_usage":{"timestamp":1738304109301,"peak_usage_bytes":73577,"status":"ok"},"analysis_stats":{"classification_stats":{"timestamp":1738304109301,"iteration":16,"hyperparameters":{"class_assignment_objective":"maximize_minimum_recall","alpha":1.8314331120379204,"downsample_factor":0.47401699894777016,"eta":0.23663627225348066,"eta_growth_rate_per_tree":1.234522328627
[truncated]
```

**Issue Reasons:**
- [8.17] 2 failures in test testWithOnlyTrainingRowsAndTrainingPercentIsHundred (0.7% fail rate in 295 executions)
- [8.17] 2 failures in pipeline elasticsearch-periodic (20.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -382,9 +382,6 @@ tests:
 - class: org.elasticsearch.xpack.security.operator.OperatorPrivilegesIT
   method: testEveryActionIsEitherOperatorOnlyOrNonOperator
   issue: https://github.com/elastic/elasticsearch/issues/119911
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsKeyword
-  issue: https://github.com/elastic/elasticsearch/issues/120071
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldSourceOnlyRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/120080
@@ -483,18 +480,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search.vectors/42_knn_search_int4_flat/Vector similarity with filter only}
   issue: https://github.com/elastic/elasticsearch/issues/121412
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testStopAndRestart
-  issue: https://github.com/elastic/elasticsearch/issues/121433
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testDependentVariableIsNested
-  issue: https://github.com/elastic/elasticsearch/issues/121458
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithDatastreams
-  issue: https://github.com/elastic/elasticsearch/issues/121236
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsBoolean
-  issue: https://github.com/elastic/elasticsearch/issues/121680
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=transform/*}
   issue: https://github.com/elastic/elasticsearch/issues/120816
@@ -504,9 +489,6 @@ tests:
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testGetProfiles
   issue: https://github.com/elastic/elasticsearch/issues/121101
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testWithOnlyTrainingRowsAndTrainingPercentIsHundred
-  issue: https://github.com/elastic/elasticsearch/issues/121368
 - class: org.elasticsearch.xpack.ml.packageloader.action.ModelLoaderUtilsTests
   method: testSplitIntoRanges
   issue: https://github.com/elastic/elasticsearch/issues/121799
@@ -523,14 +505,8 @@ tests:
   method: testHasPrivileges
   issue: https://github.com/elastic/elasticsearch/issues/121346
 - class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testDeleteExpiredData_RemovesUnusedState
-  issue: https://github.com/elastic/elasticsearch/issues/116234
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
   method: testWithOnlyTrainingRowsAndTrainingPercentIsFifty_DependentVariableIsTextAndKeyword
   issue: https://github.com/elastic/elasticsearch/issues/121502
-- class: org.elasticsearch.xpack.ml.integration.ClassificationIT
-  method: testSingleNumericFeatureAndMixedTrainingAndNonTrainingRows
-  issue: https://github.com/elastic/elasticsearch/issues/121479
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testSuggestProfilesWithName
   issue: https://github.com/elastic/elasticsearch/issues/121022

